.github/workflows/documentation.yml

```diff
@@ -0,0 +1,50 @@
+name: documentation
+
+on:
+  pull_request:
+    branches: [documentation]
+  push:
+    branches: [documentation]
+
+jobs:
+ checks:
+   if: github.event_name != 'push'
+   runs-on: ubuntu-latest
+   steps:
+     - uses: actions/checkout@v1
+     - uses: actions/setup-node@v1
+       with:
+         node-version: '12.x'
+         run: |
+          npm ci
+          npm run build
+ gh-release:
+   if: github.event_name != 'pull_request'
+   runs-on: ubuntu-latest
+   steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+    - name: Add key
+      env:
+        SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      run: |
+        mkdir -p ~/.ssh
+        ssh-keyscan github.com >> ~/.ssh/known_hosts
+        echo "${{ secrets.SERVER_DEPLOY_KEY }}" > ~/.ssh/id_rsa
+        chmod 600 ~/.ssh/id_rsa
+        cat <<EOT >> ~/.ssh/config
+        Host github.com
+        HostName github.com
+        IdentityFile ~/.ssh/id_rsa
+        EOT
+    - name: Release to GitHub Pages
+      env:
+        USE_SSH: true
+        GIT_USER: git
+      run: |
+        git config --global user.email "actions@gihub.com"
```

Comment:
Have you tried this? `actions@gihub.com` probably won't work and if it does it is probably not necessary ;)

Reply:
I tested the whole workflow in this repository: https://github.com/artemkovalyov/cloud-sdk ent-to-end. The git user, in this case, is a dummy user needed for git not to fail. It can be whatever as under the hood different user is used.

Reply:
Maybe you could make this clear by adding a comment or a description. However, it surprises me that this dummy user is necessary after all...

Reply:
Agree, this is a bit of dirty hack. But without it, git complains and asks to add credentials. It's because it is used not via actions or smth but directly in the ubuntu instance they sprout to run the action. So no bindings.